.pre-commit-config.yaml

```diff
@@ -1,10 +1,42 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
 repos:
-  - repo: https://github.com/astral-sh/ruff-pre-commit
-    # Ruff version.
+-   repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v3.2.0
+    hooks:
+    -   id: check-yaml
+        name: Check YAML file formatting
+    -   id: check-added-large-files
+        name: No large Files
+        description: Checks that you did not unconsciously upload some enormous file
+    -   id: check-ast
+        name: Check Valid Python
+        description: check if python file is valid
+    -   id: end-of-file-fixer
+        name: End of File Fix
+        description: Checks file ends on newline
+    -   id: no-commit-to-branch
+        name: No direct commit to main
+        args: ['--branch', 'main']
+        description: Do not commit to main directly
+    -   id: trailing-whitespace
+        name: Trim trailing whitespaces
+        description: Trailing whitespaces are trimmed
+-   repo: https://github.com/astral-sh/ruff-pre-commit
     rev: v0.1.11
     hooks:
-      # Run the linter.
-      - id: ruff
+    -   id: ruff
         args: [ --fix ]
-      # Run the formatter.
-      - id: ruff-format
+        description: python linter
+    -   id: ruff-format
+        description: python formatter
+-   repo: https://github.com/psf/black-pre-commit-mirror
+    rev: 24.3.0
+    hooks:
+    -   id: black-jupyter
+        description: formating ipynb notebook
+-   repo: https://github.com/compilerla/conventional-pre-commit
+    rev: v3.4.0
+    hooks:
+    -   id: conventional-pre-commit
+        stages: [commit-msg]
```

Comment:
Yes I like this one because I like (even very basic) conventional commit messages, but of course if other people find it too annoying we can always delete it.